### DISCUSSION TITLE
Is there is a way to unregister through Controller

### DISCUSSION THREAD

**Discussion by maxy4u** [react-hook-form/react-hook-form#1884] [Q&A] [answered]
I have a generic component which internally uses Controller. 
We have a requirement where we need some fields to get disabled if certain fields are populated. Perhaps this is asked before. However, in my  case i am passing conditional rules props to controller if fields is disabled or rules are empty then we are not sending rules prop to Controller

![Screen Shot 2020-06-16 at 3 50 02 PM](https://user-images.githubusercontent.com/6682288/84815209-169f4980-afe9-11ea-9a36-357dc21e0f77.png)

Its appears rules once passed are cached and stays with controller. Is there a way to Unregister or Override Validate ? 

I tried to add a condition where in if field is disabled we will return new validate with required function to return true (hope return true means there will be no error displayed for that field ). But this validate who return true never gets called i tried returning false as well it seems initial required gets called always when controller was registered at the first time.
Note:- I know we can used getValues in validate but we need a different solution as our validates are generic .

Thanks in advance. :)

**Comment by bluebill1049** [comment]
This is fixed on v6 (coming) :) for workaround checkout this issue: https://github.com/react-hook-form/react-hook-form/issues/1749

**Reply by maxy4u** [reply]
HI Thanks for replying :) 
I was going through the workaround. However, unable to see the logic to unregister Controlled component. 
Am i missing anything?

Thank you once again for nice library :)

**Comment by maxy4u** [comment]
@bluebill1049  created a sample code scenario for the same. Do you mind helping me determine how to unregister controlled field.
https://stackblitz.com/edit/react-dnbcsc

**Comment by maxy4u** [comment, marked as answer]
@bluebill1049  Never mind i figured out the way updated my code to use 
const key = `${ind}-${+(disabled || false)}` and passed it to controller. Thanks for above link

**Reply by SahDhiraj** [reply]
didn't get it,
do you mean you added key prop in the controller ?
